### PR TITLE
Refactored to use function components and tidied formatting and spacing

### PR DESCRIPTION
**Pull request by ChangoFoster**
In theory everything should work the same after this as far as I can tell. I'm not that familiar with the getInitialProps function from next so might be worth checking that does the same thing.

---
This change is [Reviewable](https://reviewable.io/reviews/jamestwisleton/musicjournal/2)

### CODE REVIEW

File: music-journal/components/ListMemories.js [hunk 1]
```diff
@@ -1,68 +1,33 @@
...
-      songColumn.innerHTML = memories[i].song;
-      memoryColumn.innerHTML = memories[i].text;
-      tableRow.appendChild(songColumn);
-      tableRow.appendChild(memoryColumn);
-      tableBody.appendChild(tableRow);
-    }
-  }
-
-  componentDidMount() {
+  useEffect(() => {
```
Comment (JamesTwisleton) [inline music-journal/components/ListMemories.js:13]: hope this works, think I _may_ have experimented with useEffect and couldn't get it to work

File: music-journal/src/initFirebase.js [hunk 2]
```diff
@@ -2,18 +2,21 @@ import firebase from 'firebase/app';
 import 'firebase/auth';
 
 const config = {
-apiKey: process.env.FIREBASE_API_KEY,
-authDomain: process.env.FIREBASE_AUTH_DOMAIN,
-databaseURL: process.env.FIREBASE_DATABASE_URL,
-projectId: process.env.FIREBASE_PROJECT_ID,
-storageBucket: process.env.FIREBASE_STORAGE_BUCKET,
-messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID
+    apiKey: process.env.FIREBASE_API_KEY,
```
Comment (JamesTwisleton) [inline music-journal/src/initFirebase.js:5]: made my heart jump here, thought for a second you'd committed the credentials 😹

File: music-journal/src/initFirebase.js [hunk 3]
```diff
@@ -2,18 +2,21 @@ import firebase from 'firebase/app';
...
 }
+
 const auth = firebase.auth();
+
 export {
- auth,
- firebase
+    auth,
+    firebase
 };
```
Comment (JamesTwisleton) [inline music-journal/src/initFirebase.js:22]: I guess address the missing newlines at end of a few of these files :)

File: music-journal/components/ListMemories.js (hunk 1 above)
Comment (ChangoFoster) [reply music-journal/components/ListMemories.js:13]: the main thing with useEffect is the second argument which lets it know when to run the function again. If you have an empty array it just does it when the component loads. Looks like it works.

File: music-journal/src/initFirebase.js (hunk 2 above)
Comment (ChangoFoster) [reply music-journal/src/initFirebase.js:5]: Should just be formatting things rather than creds!

File: music-journal/src/initFirebase.js (hunk 3 above)
Comment (ChangoFoster) [reply music-journal/src/initFirebase.js]: Done!

File: music-journal/components/ListMemories.js (hunk 1 above)
Comment (JamesTwisleton) [reply music-journal/components/ListMemories.js:13]: TIL!

**Pull request merged**